### DISCUSSION TITLE
Remove dependency on Indy

### DISCUSSION THREAD

**Discussion by MHumm** [AliDehbansiahkarbon/ChatGPTWizard#25] [Announcements]
In order to reduce the dependencies on 3rd party libraries for this project and on OpenSSL DLLs it would be nice if the REST requests needed could be done using either System.Net.HttpClientComponent.TNetHTTPClient or System.Net.HttpClient.THTTPClient.

These wouldn't need OpenSSL DLLs for TLS as they'd use the Windows APIs and thus Windows built in TLS support.

**Comment by AliDehbansiahkarbon** [comment]
Well, try the latest version, please.
This is not an issue so I transferred it here to the discussions page. 

There are two reasons for that.
1- System.Net.HttpClient.THTTPClient is not available before XE8 (if I'm not wrong) and the idea behind using Indy was to support some older IDE versions at that time.

2- Sometimes the response is not a valid JSON, using the built-in library, and works fine with Indy, I couldn't really figure out why.

Anyway, I changed my mind recently and made some changes in this regard, in newer IDEs (Berlin and later) the plug-in will try to use built-in HTTPClient first, and try with Indy only if the result of the first method is not a valid response or JSON or anything goes wrong.
So using the newer IDEs you shouldn't need SSL libraries unless something goes wrong with the built-in library.

**Reply by CCH44BI** [reply]
Seems to work in D10.4 Sidney CE. renamed the 2 DLL from your resource folder to something else

**Comment by AliDehbansiahkarbon** [comment]
I just noticed that THTTPClient doesn't have some critical properties like ConnectionTimeout and ResponseTimeout even in Seattle!
It's not reliable at all in the older IDE versions than Rio it seems, so I have to keep Indy for those IDEs.

**Reply by MHumm** [reply]
This might well be that these old version do not provide all relevant properties. But to be honest: why do we need to support such old IDE versions? Is there so much demand?

**Reply by AliDehbansiahkarbon** [reply]
I will do that at the end for sure, but for now I have to mention that we did a poll in Delphi community about the versions that are actively used, and based on the result I don't want to ignore a big part of the community as it's possible.
Also this plugin is open source so you can customize it in any aspect.
